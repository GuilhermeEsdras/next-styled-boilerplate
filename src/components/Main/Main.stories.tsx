import { Story, Meta } from '@storybook/react/types-6-0';

import Main from './Main';

const meta: Meta = {
  title: 'Main',
  component: Main,
  args: {
    title: 'Default title',
    description: 'Default description',
  },
};

export const Basic: Story = (args) => <Main {...args} />;
Basic.args = {
  title: 'Basic title',
  description: 'Basic description',
};

export const Default: Story = (args) => <Main {...args} />;

export default meta;
