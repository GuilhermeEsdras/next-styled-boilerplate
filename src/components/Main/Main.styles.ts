import styled, { css } from 'styled-components';
import media from 'styled-media-query';

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background: ${theme.colors.background};
    color: #fff;
    width: 100%;
    height: 100%;
    padding: 3rem;
    text-align: center;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    ${media.lessThan('medium')`
      width: 100%;
    `}
  `}
`;

export const Logo = styled.img`
  width: 15rem;
  margin-bottom: 2rem;
`;

export const Title = styled.h1`
  font-size: 2.5rem;
`;

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`;

export const Illustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`;
