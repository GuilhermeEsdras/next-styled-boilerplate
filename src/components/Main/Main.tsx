const Main = function () {
  return (
    <main>
      <h2>Esdras proj initialzer</h2>
    </main>
  );
};

export default Main;
