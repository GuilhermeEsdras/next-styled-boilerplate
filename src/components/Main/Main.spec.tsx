import renderer from 'react-test-renderer';

import Main from './Main';

it('renders Main unchanged', () => {
  const tree = renderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
});
