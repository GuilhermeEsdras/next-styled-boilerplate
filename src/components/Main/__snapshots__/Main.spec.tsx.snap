// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Main unchanged 1`] = `
<main>
  <h2>
    Esdras proj initialzer
  </h2>
</main>
`;
